libs/core/langchain_core/retrievers.py

```diff
@@ -22,8 +24,19 @@
         Callbacks,
     )
 
+
+class DocumentResult(Serializable):
+    document: Document
+    metadata: dict = Field(default_factory=dict)
+
+
+class RetrievalResult(Serializable):
```

Comment:
This API works IMO

Reply:
any argument to be made that we should explicitly include `query: str`?